src/vs/workbench/services/editor/browser/editorResolverService.ts

```diff
@@ -255,7 +255,9 @@ export class EditorResolverService extends Disposable implements IEditorResolver
 
 	getAssociationsForResource(resource: URI): EditorAssociations {
 		const associations = this.getAllUserAssociations();
-		const matchingAssociations = associations.filter(association => association.filenamePattern && globMatchesResource(association.filenamePattern, resource));
+		let matchingAssociations = associations.filter(association => association.filenamePattern && globMatchesResource(association.filenamePattern, resource));
+		// Sort matching associations based on glob length as a longer glob will be more specific
+		matchingAssociations = matchingAssociations.sort((a, b) => (b.filenamePattern?.length ?? 0) - (a.filenamePattern?.length ?? 0));
 		const allEditors: RegisteredEditors = this._registeredEditors;
 		// Ensure that the settings are valid editors
 		return matchingAssociations.filter(association => allEditors.find(c => c.editorInfo.id === association.viewType));
```

Comment:
Not sure if this is run frequently, but if it is you would be better off with a for const loop to avoid creating a new array per filter/sort call

Reply:
I'm not sure what you mean by this? How would you use a fore const loop? I could write this as one changed thing but under the hood I think it's the same

```
const matchingAssociations = associations
  .filter(association => association.filenamePattern && globMatchesResource(association.filenamePattern, resource))
  .sort((a, b) => (b.filenamePattern?.length ?? 0) - (a.filenamePattern?.length ?? 0))
  .filter(association => allEditors.find(c => c.editorInfo.id === association.viewType));
```